### PR TITLE
Add autofix to case-*

### PR DESCRIPTION
**Pull request by luoxlgh**
Adding auto fix to existing rules: property-case, unit-case, value-keyword-case, selector-type-case, selector-pseudo-element-case and selector-pseudo-class-case

### CODE REVIEW

File: lib/rules/selector-pseudo-element-case/__tests__/index.js [hunk 1]
```diff
@@ -95,114 +96,91 @@ testRule(rule, {
...
-      message: messages.expected("::First-letter", "::first-letter"),
-      line: 1,
-      column: 2
-    },
-    {
-      code: "a::First-line { color: pink; }",
-      message: messages.expected("::First-line", "::first-line"),
-      line: 1,
-      column: 2
-    },
```
Comment (hudochenkov) [inline lib/rules/selector-pseudo-element-case/__tests__/index.js]: Put these tests back, please.

File: lib/rules/selector-pseudo-element-case/__tests__/index.js [hunk 2]
```diff
@@ -353,114 +340,91 @@ testRule(rule, {
...
-      message: messages.expected("::First-letter", "::FIRST-LETTER"),
-      line: 1,
-      column: 2
-    },
-    {
-      code: "a::First-line { color: pink; }",
-      message: messages.expected("::First-line", "::FIRST-LINE"),
-      line: 1,
-      column: 2
-    },
```
Comment (hudochenkov) [inline lib/rules/selector-pseudo-element-case/__tests__/index.js]: Put these tests back, please.

**Comment by jeddy3** [comment]
@luoxlgh A big thank you for contributing this, especially as I'm sure it took quite a while to populated all the `fixed` values!

Can you add back in those tests that @hudochenkov [requested](https://github.com/stylelint/stylelint/pull/3080#discussion_r158696389). I think once those are back in, this is PR is looking good to me.

**Comment by jeddy3** [comment]
@luoxlgh Thanks for adding the tests back in.

File: lib/rules/property-case/README.md [hunk 3]
```diff
@@ -8,6 +8,8 @@ Specify lowercase or uppercase for properties.
  * These properties */
 ```
 
+The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix some of the problems reported by this rule.
```
Comment (hudochenkov) [inline lib/rules/property-case/README.md]: ~~some~~ all:

```md
The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix all of the problems reported by this rule.
```

File: lib/rules/selector-pseudo-class-case/README.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ Specify lowercase or uppercase for pseudo-class selectors.
  * This is pseudo-class selector */
 ```
 
+The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix some of the problems reported by this rule.
```
Comment (hudochenkov) [inline lib/rules/selector-pseudo-class-case/README.md]: ~~some~~ all:

```md
The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix all of the problems reported by this rule.
```

File: lib/rules/selector-type-case/README.md [hunk 5]
```diff
@@ -8,6 +8,8 @@ Specify lowercase or uppercase for type selectors.
  * This is type selector */
 ```
 
+The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix some of the problems reported by this rule.
```
Comment (hudochenkov) [inline lib/rules/selector-type-case/README.md]: ~~some~~ all:

```md
The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix all of the problems reported by this rule.
```

File: lib/rules/unit-case/README.md [hunk 6]
```diff
@@ -8,6 +8,8 @@ Specify lowercase or uppercase for units.
  *     These units */
 ```
 
+The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix some of the problems reported by this rule.
```
Comment (hudochenkov) [inline lib/rules/unit-case/README.md]: ~~some~~ all:

```md
The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix all of the problems reported by this rule.
```

File: lib/rules/value-keyword-case/README.md [hunk 7]
```diff
@@ -10,6 +10,8 @@ Specify lowercase or uppercase for keywords values.
 
 This rule ignores [`<custom-idents>`](https://developer.mozilla.org/en/docs/Web/CSS/custom-ident) of known properties. Values which are paired with non-properties (e.g. `$vars` and custom properties), and do not conform to the primary option, can be ignored using the `ignoreValues: []` secondary option.
 
+The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix some of the problems reported by this rule.
```
Comment (hudochenkov) [inline lib/rules/value-keyword-case/README.md]: ~~some~~ all:

```md
The `--fix` option on the [command line](../../../docs/user-guide/cli.md#autofixing-errors) can automatically fix all of the problems reported by this rule.
```

**Comment by jeddy3** [comment]
> There is a problem with how fix applied in selector-type-case, unit-case, and value-keyword-case. Because fix applies by .replace(),

I totally didn't see that. Thanks for thinking it through, @hudochenkov. I guess `replace`s on entire params and values should be a red flag as they will always cause these types of issues.

**Comment by jeddy3** [comment]
@luoxlgh Friendly ping to see if you're still working on this PR?

**Comment by yepninja** [comment]
Already solved #3725

**Pull request closed without merging**